### PR TITLE
Add recipe for the ACE (ADAPTIVE Communication Environment)

### PR DESCRIPTION
**Pull request by traversaro**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

The ACE (ADAPTIVE Communication Environment) is a C++ library that implements many core patterns for concurrent communication software.

As far as I was able to test, the recipe should be ready, so I guess I can tag @conda-forge/help-c-cpp as this is a recipe for a C++ port, and perhaps also @wolfv as it is a recipe loosely related to robotics (I am packaging ACE as, on Windows, it is a dependency for the [YARP robotics middleware](http://yarp.it/) used in the [iCub humanoid robot](https://icub.iit.it/)). 

Fix https://github.com/conda-forge/staged-recipes/issues/10707 .

### CODE REVIEW

**Comment by traversaro** [comment]
> License is not an SPDX identifier (or a custom LicenseRef) nor an SPDX license expression.

Thanks for noticing. Apparently ACE has its own BSD-like license, that fortunately has its own SPDX identified ([`DOC`](https://spdx.org/licenses/DOC.html)).

**Comment by traversaro** [comment]
> As far as I was able to test, the recipe should be ready, so I guess I can tag @conda-forge/help-c-cpp as this is a recipe for a C++ port, and perhaps also @wolfv as it is a recipe loosely related to robotics (I am packaging ACE as, on Windows, it is a dependency for the [YARP robotics middleware](http://yarp.it/) used in the [iCub humanoid robot](https://icub.iit.it/)).

Obviously I was a bit too optimistic, as the recipe is not compiling on the CI.

**Comment by wolfv** [comment]
awesome! Looks like you're missing perl :) You can add it to build (or host) with `- perl`, it should be available on all platforms.

**Comment by traversaro** [comment]
> awesome! Looks like you're missing perl :) You can add it to build (or host) with `- perl`, it should be available on all platforms.

Thanks, but indeed perl is used before in the build script, and is already installed (see https://github.com/conda-forge/staged-recipes/pull/12698/files#diff-30cdf92a5e975ac211bed07ca5ad2e08R21). However I guess that for some reason the error on Linux is: 
~~~
make[1]: Leaving directory '$SRC_DIR/ace/Monitor_Control'
make[1]: Entering directory '$SRC_DIR/ace'

GNUmakefile: $SRC_DIR/ace/GNUmakefile.ACE MAKEFLAGS=w

perl -ne'if (/^#END MPC-Generated Install Info/) {exit 0}' \
  -e'elsif (/^#BEGIN MPC-Generated Install Info/) {$ok=1}' \
  -e'elsif ($ok && s/^#//) {print}' GNUmakefile.ACE | \
$SRC_DIR/MPC/prj_install.pl -i -s lib_output,header_files,template_files,inline_files,installdata_files \
  -b header_files=include/ace -b template_files=include/ace -b inline_files=include/ace -o lib_output=lib -x installdata_files -b installdata_files=share/ace -d 64  \
  $PREFIX
/bin/sh: /home/conda/staged-recipes/build_artifacts/ace_1600618214199/work/MPC/prj_install.pl: /usr/bin/perl: bad interpreter: No such file or directory
make[1]: *** [GNUmakefile.ACE:530: install] Error 126
make[1]: Leaving directory '$SRC_DIR/ace'
make: *** [GNUmakefile:31: install] Error 2
~~~
I had the same error locally, and I fixed it by switching to the `.tar.gz` archive as suggested in https://github.com/microsoft/vcpkg/blame/2020.07/ports/ace/portfile.cmake#L1, but probably there is something that I am missing that is going on. @jwillemsen just in case, do you have any idea on this failure? 

On the other hand, on macOS the failure is:
~~~
2020-09-20T16:11:57.9115640Z Generation Time: 3s
2020-09-20T16:11:57.9782280Z make[1]: Entering directory '$SRC_DIR/ace'
2020-09-20T16:12:05.3577530Z touch .depend.ACE
2020-09-20T16:12:05.5375790Z 
2020-09-20T16:12:05.5402570Z GNUmakefile: $SRC_DIR/ace/GNUmakefile.ACE MAKEFLAGS=w -j2 --jobserver-auth=4,5
2020-09-20T16:12:05.5421940Z 
2020-09-20T16:12:05.5529880Z clang++ -fdelayed-template-parsing -g -m64 -W -Wall -D_FORTIFY_SOURCE=2 -mmacosx-version-min=10.9 -isystem $PREFIX/include -DACE_HAS_CUSTOM_EXPORT_MACROS=0  -I$SRC_DIR -D__ACE_INLINE__ -I.. -DACE_USES_WCHAR -DACE_BUILD_DLL  -c  -o 64/.shobj/ACE_crc32.o $SRC_DIR/ace/ACE_crc32.cpp
2020-09-20T16:12:05.5543170Z clang++ -fdelayed-template-parsing -g -m64 -W -Wall -D_FORTIFY_SOURCE=2 -mmacosx-version-min=10.9 -isystem $PREFIX/include -DACE_HAS_CUSTOM_EXPORT_MACROS=0  -I$SRC_DIR -D__ACE_INLINE__ -I.. -DACE_USES_WCHAR -DACE_BUILD_DLL  -c  -o 64/.shobj/ACE.o $SRC_DIR/ace/ACE.cpp
2020-09-20T16:12:05.6113070Z In file included from /Users/runner/miniforge3/conda-bld/ace_1600618232852/work/ace/ACE.cpp:1:
2020-09-20T16:12:05.6196130Z In file included from /Users/runner/miniforge3/conda-bld/ace_1600618232852/work/ace/ACE.h:29:
2020-09-20T16:12:05.6205660Z In file included from /Users/runner/miniforge3/conda-bld/ace_1600618232852/work/ace/Basic_Types.h:52:
2020-09-20T16:12:05.6214900Z In file included from /Users/runner/miniforge3/conda-bld/ace_1600618232852/work/ace/os_include/os_limits.h:25:
2020-09-20T16:12:05.6223550Z In file included from /Users/runner/miniforge3/conda-bld/ace_1600618232852/work/ace/os_include/os_unistd.h:25:
2020-09-20T16:12:05.6326170Z /Users/runner/miniforge3/conda-bld/ace_1600618232852/work/ace/os_include/sys/os_types.h:119:15: error: typedef redefinition with different types ('int' vs '__darwin_ssize_t' (aka 'long'))
2020-09-20T16:12:05.6379610Z   typedef int ssize_t;
2020-09-20T16:12:05.6423140Z               ^
2020-09-20T16:12:05.6424970Z /Applications/Xcode_11.7.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.9.sdk/usr/include/sys/_types/_ssize_t.h:30:33: note: previous definition is here
2020-09-20T16:12:05.6494140Z typedef __darwin_ssize_t        ssize_t; 
2020-09-20T16:12:05.6499300Z                                 ^
2020-09-20T16:12:05.6501520Z In file included from /Users/runner/miniforge3/conda-bld/ace_1600618232852/work/ace/ACE_crc32.cpp:1:
2020-09-20T16:12:05.6503440Z In file included from /Users/runner/miniforge3/conda-bld/ace_1600618232852/work/ace/ACE.h:29:
2020-09-20T16:12:05.6505280Z In file included from /Users/runner/miniforge3/conda-bld/ace_1600618232852/work/ace/Basic_Types.h:52:
2020-09-20T16:12:05.6507110Z In file included from /Users/runner/miniforge3/conda-bld/ace_1600618232852/work/ace/os_include/os_limits.h:25:
2020-09-20T16:12:05.6508970Z In file included from /Users/runner/miniforge3/conda-bld/ace_1600618232852/work/ace/os_include/os_unistd.h:25:
2020-09-20T16:12:05.6511010Z /Users/runner/miniforge3/conda-bld/ace_1600618232852/work/ace/os_include/sys/os_types.h:119:15: error: typedef redefinition with different types ('int' vs '__darwin_ssize_t' (aka 'long'))
2020-09-20T16:12:05.6512090Z   typedef int ssize_t;
2020-09-20T16:12:05.6512650Z               ^
2020-09-20T16:12:05.6517210Z /Applications/Xcode_11.7.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.9.sdk/usr/include/sys/_types/_ssize_t.h:30:33: note: previous definition is here
2020-09-20T16:12:05.6518930Z typedef __darwin_ssize_t        ssize_t; 
2020-09-20T16:12:05.6520160Z                                 ^
2020-09-20T16:12:05.6522490Z In file included from /Users/runner/miniforge3/conda-bld/ace_1600618232852/work/ace/ACE.cpp:1:
2020-09-20T16:12:05.6525390Z In file included from /Users/runner/miniforge3/conda-bld/ace_1600618232852/work/ace/ACE.h:29:
2020-09-20T16:12:05.6527850Z In file included from /Users/runner/miniforge3/conda-bld/ace_1600618232852/work/ace/Basic_Types.h:54:
2020-09-20T16:12:05.6530210Z In file included from /Users/runner/miniforge3/conda-bld/ace_1600618232852/work/ace/os_include/os_stdlib.h:27:
2020-09-20T16:12:05.6532680Z In file included from /Users/runner/miniforge3/conda-bld/ace_1600618232852/work/ace/os_include/sys/os_wait.h:25:
2020-09-20T16:12:05.6535110Z In file included from /Users/runner/miniforge3/conda-bld/ace_1600618232852/work/ace/os_include/os_signal.h:211:
2020-09-20T16:12:05.6537570Z In file included from /Users/runner/miniforge3/conda-bld/ace_1600618232852/work/ace/os_include/os_ucontext.h:28:
2020-09-20T16:12:05.6539420Z /Applications/Xcode_11.7.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.9.sdk/usr/include/ucontext.h:43:2: error: The deprecated ucontext routines require _XOPEN_SOURCE to be defined
2020-09-20T16:12:05.6541250Z #error The deprecated ucontext routines require _XOPEN_SOURCE to be defined
2020-09-20T16:12:05.6542460Z  ^
~~~

that sounds like that the wrong `config.h` was used, for some reason.

**Comment by traversaro** [comment]
The Linux error may be related to the use of the system perl, instead of the conda-provided one. I tried to pass `PERL_PATH` to make sure that the conda one was used, but apparently the ACE build system still preferred the system one. Perhaps everything is working for me locally as the system perl is new enough/similar enough to the conda provided one. 

Per macOS, I tried to debug quickly the issue using https://github.com/fastai/fastmac, and from there it seems that the correct config is used, but then the compilation fails again due to the following error:
~~~
In file included from /Users/runner/miniconda3/conda-bld/debug_1600721723474/work/ace/ACE_crc_ccitt.cpp:1:
In file included from /Users/runner/miniconda3/conda-bld/debug_1600721723474/work/ace/ACE.h:23:
In file included from /Users/runner/miniconda3/conda-bld/debug_1600721723474/work/ace/config-lite.h:22:
In file included from /Users/runner/miniconda3/conda-bld/debug_1600721723474/work/ace/config-macros.h:22:
In file included from /Users/runner/miniconda3/conda-bld/debug_1600721723474/work/ace/config.h:1:
/Users/runner/miniconda3/conda-bld/debug_1600721723474/work/ace/config-macosx.h:3:10: fatal error: 'Availability.h' file not found
#include <Availability.h>
         ^~~~~~~~~~~~~~~~
In file included from /Users/runner/miniconda3/conda-bld/debug_1600721723474/work/ace/ATM_Acceptor.cpp:1:
In file included from /Users/runner/miniconda3/conda-bld/debug_1600721723474/work/ace/ATM_Acceptor.h:16:
In file included from /Users/runner/miniconda3/conda-bld/debug_1600721723474/work/ace/config-all.h:18:
In file included from /Users/runner/miniconda3/conda-bld/debug_1600721723474/work/ace/config-lite.h:22:
In file included from /Users/runner/miniconda3/conda-bld/debug_1600721723474/work/ace/config-macros.h:22:
In file included from /Users/runner/miniconda3/conda-bld/debug_1600721723474/work/ace/config.h:1:
/Users/runner/miniconda3/conda-bld/debug_1600721723474/work/ace/config-macosx.h:3:10: fatal error: 'Availability.h' file not found
#include <Availability.h>
         ^~~~~~~~~~~~~~~~
1 error generated.
~~~

**Comment by jwillemsen** [comment]
Are you sure that the config.h file is created correctly for macosx, the errors shown are very basic. 

The perl issue on linux could be line ending, but you do have the tar.gz, maybe some file is not there?

File: recipes/ace/build.sh [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+export ACE_SOURCE_PATH=${ACE_ROOT}/ace
+export WORKSPACE=${ACE_ROOT}/ace/ace
+export INSTALL_PREFIX=$PREFIX
+export PERL_PATH=$CONDA_PREFIX/bin/perl
+
+# Configure step
+cd ${ACE_ROOT}
+perl ${ACE_ROOT}/bin/mwc.pl -type gnuace -features "uses_wchar=1,zlib=0,ssl=0,openssl11=0,trio=0,xt=0,fl=0,fox=0,tk=0,qt=0,rapi=0,stlport=0,rwho=0" ${WORKSPACE}.mwc
+
+if [ -z "$MACOSX_DEPLOYMENT_TARGET" ]
```
Comment (jwillemsen) [inline recipes/ace/build.sh:15]: Is this the correct check?

File: recipes/ace/bld.bat [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+REM Configure step
+cd %ACE_ROOT%
+perl %ACE_ROOT%\bin\mwc.pl -type vs2017 -features "uses_wchar=1,zlib=0,ssl=0,openssl11=0,trio=0,xt=0,fl=0,fox=0,tk=0,qt=0,rapi=0,stlport=0,rwho=0" %WORKSPACE%.mwc
+
+REM Create config.h file
+echo #include "ace/config-windows.h" > %ACE_SOURCE_PATH%\config.h
+
+REM Build step
+echo "Executing msbuild %SLN_FILE% /p:Configuration=%SLN_CFG%,Platform=%SLN_PLAT%,PlatformToolset=v141"
+msbuild %SLN_FILE% /p:Configuration=%SLN_CFG%,Platform=%SLN_PLAT%,PlatformToolset=v141
```
Comment (jwillemsen) [inline recipes/ace/bld.bat:25]: Add -maxcpucount to let msbuild use all cpu cores

File: recipes/ace/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,36 @@
+{% set name = "ace" %}
+{% set version = "6.5.11" %}
+
+package:
+  name: {{ name }}
+  version: {{ version }}
+
+source:
+  - url: https://github.com/DOCGroup/ACE_TAO/releases/download/ACE%2BTAO-6_5_11/ACE-src-6.5.11.tar.gz
```
Comment (jwillemsen) [inline recipes/ace/meta.yaml:9]: Can you use version here?

File: recipes/ace/bld.bat [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+
+echo %SRC_DIR%
+echo %ACE_ROOT%
+echo %PREFIX%
+echo %LIBRARY_INC%
+echo %SLN_PLAT%
+
+REM Configure step
+cd %ACE_ROOT%
+perl %ACE_ROOT%\bin\mwc.pl -type vs2017 -features "uses_wchar=1,zlib=0,ssl=0,openssl11=0,trio=0,xt=0,fl=0,fox=0,tk=0,qt=0,rapi=0,stlport=0,rwho=0" %WORKSPACE%.mwc
```
Comment (jwillemsen) [inline recipes/ace/bld.bat:18]: What about vs2019?

**Comment by jwillemsen** [comment]
A search (https://github.com/search?q=conan+ace) did show some conan python recipes for ACE, maybe something to look at?

File: recipes/ace/bld.bat (hunk 4 above)
Comment (wolfv) [reply recipes/ace/bld.bat:18]: we only have vs2017 on conda-forge afaik.

File: recipes/ace/build.sh (hunk 1 above)
Comment (traversaro) [reply recipes/ace/build.sh:15]: I checked for the environment variables that are only defined on macOS, I do not know if some conda expert (@wolfv @Tobias-Fischer ?) can suggest a better check to have conditional code only on macOS.

Comment (wolfv) [reply recipes/ace/build.sh:15]: there is the `$target_platform` env var which would be `osx-64`, or `osx-arm64`, but many people also chekc using `uname`. 

A real life-saver is the org-wide search on github: https://github.com/search?q=org%3Aconda-forge+target_platform&type=code

MACOSX_DEPLOYMENT_TARGET might also soon be deprecated (or mabye already is :/)

**Comment by wolfv** [comment]
locally on linux this recipe built fine ... let's give it another try

**Comment by traversaro** [comment]
> 
> 
> locally on linux this recipe built fine ... let's give it another try

On which distro? As mentioned in https://github.com/conda-forge/staged-recipes/pull/12698#issuecomment-696379694, I guess there is some confusion in use between distro and conda perl, and perhaps if they are similar enough everything works.

**Comment by wolfv** [comment]
Fedora 32

one other solution might be to forcibly remove the distro's perl?

**Comment by wolfv** [comment]
The OS X error is legitimate? We can skip os x by adding 

```
build:
  skip: true  # [osx]
```

**Comment by wolfv** [comment]
I tried some stuff, let's see if this goes better :)

**Comment by wolfv** [comment]
I think the issue is that somewhere in the linux scripts `/usr/bin/perl` is hardcoded instead of using `/usr/bin/env perl` or something like that at the top of some script. Since there is no perl available on the dokcer image it fails.

Now -- we could install a perl with yum (bad solution), I tried to create a symbolic link (also didn't work because no sudo permission, or we can patch the build system...)

**Comment by jwillemsen** [comment]
Made an upstream PR for using `env perl`

**Comment by wolfv** [comment]
ah now my dirty fix fails. thanks for the fix @jwillemsen

**Comment by jwillemsen** [comment]
> ah now my dirty fix fails. thanks for the fix @jwillemsen

Will need to go through review, CI, and a new release, no schedule for a new ACE release yet

**Comment by wolfv** [comment]
I was trying to say that the build failed because of my attempt to use `ln -s /usr/bin/perl` :)

I think your patch will work, let's see!

Maybe we want to `cat config.h` to see the contents in the CI? I don't know what's going on on OS X...

**Comment by wolfv** [comment]
looks like there is (at least one) another perl file to fix: `work/MPC/prj_install.pl`

**Comment by jwillemsen** [comment]
> looks like there is (at least one) another perl file to fix: `work/MPC/prj_install.pl`

https://github.com/DOCGroup/MPC/pull/106

**Comment by jwillemsen** [comment]
> looks like there is (at least one) another perl file to fix: `work/MPC/prj_install.pl`

You could try the workaround to add the following line to the top of the platform_macros.GNU file that gets created
```
INSTALLER = perl $(MPC_ROOT)/prj_install.pl
```

**Comment by wolfv** [comment]
I can add your patches. It should be straightforward.

Can you point me to teh file that we should `cat` and at what stage of the buidl process?

File: recipes/ace/build.sh [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+cd ${ACE_ROOT}
+perl ${ACE_ROOT}/bin/mwc.pl -type gnuace -features "uses_wchar=1,zlib=0,ssl=0,openssl11=0,trio=0,xt=0,fl=0,fox=0,tk=0,qt=0,rapi=0,stlport=0,rwho=0" ${WORKSPACE}.mwc
+
+if [[ $target_platform == osx* ]]
+then
+  printf "#include \"ace/config-macosx.h\"" > ${ACE_SOURCE_PATH}/config.h
+  printf "include ${ACE_ROOT}/include/makeinclude/platform_macosx.GNU" > ${ACE_ROOT}/include/makeinclude/platform_macros.GNU
+else
+  printf "#include \"ace/config-linux.h\"" > ${ACE_SOURCE_PATH}/config.h
+  printf "include ${ACE_ROOT}/include/makeinclude/platform_linux.GNU" > ${ACE_ROOT}/include/makeinclude/platform_macros.GNU
```
Comment (jwillemsen) [inline recipes/ace/build.sh:21]:
``` suggestion
printf "INSTALLER = perl $(MPC_ROOT)/prj_install.pl" > ${ACE_ROOT}/include/makeinclude/platform_macros.GNU
printf "include ${ACE_ROOT}/include/makeinclude/platform_linux.GNU" >> ${ACE_ROOT}/include/makeinclude/platform_macros.GNU

File: recipes/ace/build.sh [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+fi
+
```
Comment (jwillemsen) [inline recipes/ace/build.sh:23]:
``` suggestion 
cat ${ACE_SOURCE_PATH}/config.h
cat ${ACE_ROOT}/include/makeinclude/platform_macros.GNU
```

File: recipes/ace/bld.bat (hunk 2 above)
Comment (jwillemsen) [inline recipes/ace/bld.bat:25]:
``` suggestion
msbuild %SLN_FILE% /p:Configuration=%SLN_CFG%,Platform=%SLN_PLAT%,PlatformToolset=v141 /maxcpucount
```

File: recipes/ace/build.sh [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+# Configure step
+cd ${ACE_ROOT}
+perl ${ACE_ROOT}/bin/mwc.pl -type gnuace -features "uses_wchar=1,zlib=0,ssl=0,openssl11=0,trio=0,xt=0,fl=0,fox=0,tk=0,qt=0,rapi=0,stlport=0,rwho=0" ${WORKSPACE}.mwc
+
+if [[ $target_platform == osx* ]]
+then
+  echo "Detected OS X"
+  printf "#include \"ace/config-macosx.h\"" > ${ACE_SOURCE_PATH}/config.h
+  printf "include ${ACE_ROOT}/include/makeinclude/platform_macosx.GNU" > ${ACE_ROOT}/include/makeinclude/platform_macros.GNU
```
Comment (jwillemsen) [inline recipes/ace/build.sh:19]:
``` suggestion
 printf "include $(ACE_ROOT)/include/makeinclude/platform_macosx.GNU" > ${ACE_ROOT}/include/makeinclude/platform_macros.GNU
```

File: recipes/ace/build.sh [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+else
+  echo "Detected Linux"
+  printf "#include \"ace/config-linux.h\"" > ${ACE_SOURCE_PATH}/config.h
+  printf "INSTALLER = perl $(MPC_ROOT)/prj_install.pl" > ${ACE_ROOT}/include/makeinclude/platform_macros.GNU
+  printf "include ${ACE_ROOT}/include/makeinclude/platform_linux.GNU" >> ${ACE_ROOT}/include/makeinclude/platform_macros.GNU
```
Comment (jwillemsen) [inline recipes/ace/build.sh:24]:
``` suggestion
  printf "include $(ACE_ROOT)/include/makeinclude/platform_linux.GNU" >> ${ACE_ROOT}/include/makeinclude/platform_macros.GNU
```

File: recipes/ace/build.sh [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+
+if [[ $target_platform == osx* ]]
+then
+  echo "Detected OS X"
+  printf "#include \"ace/config-macosx.h\"" > ${ACE_SOURCE_PATH}/config.h
+  printf "include ${ACE_ROOT}/include/makeinclude/platform_macosx.GNU" > ${ACE_ROOT}/include/makeinclude/platform_macros.GNU
+else
+  echo "Detected Linux"
+  printf "#include \"ace/config-linux.h\"" > ${ACE_SOURCE_PATH}/config.h
+  printf "include ${ACE_ROOT}/include/makeinclude/platform_linux.GNU" > ${ACE_ROOT}/include/makeinclude/platform_macros.GNU
```
Comment (jwillemsen) [inline recipes/ace/build.sh:23]:
``` suggestion
  printf "include $(ACE_ROOT)/include/makeinclude/platform_linux.GNU" > ${ACE_ROOT}/include/makeinclude/platform_macros.GNU
```

**Comment by wolfv** [comment]
Linux seems to be fine to me now ... should we disable os x for now?

**Comment by jwillemsen** [comment]
Can you try my suggestion with ()?

**Comment by jwillemsen** [comment]
Ok, looks the MacOSX detection doesn't work correctly, see https://github.com/jwillemsen/staged-recipes/tree/add-ace/recipes/ace and for what I managed to get working

**Comment by traversaro** [comment]
> Can you try my suggestion with ()?

Unless I did something wrong, it seems that using () creates a failure even on Linux.

**Comment by jwillemsen** [comment]
See my pr to your branch

**Comment by wolfv** [comment]
@jwillemsen maybe you want to be a recipe maintainer, too? That would give you merge and commit rights on the to-be-created feedstock (once the recipe is merged from here).

**Comment by jwillemsen** [comment]
Yes @wolfv! That way I can update the recipe with a new ace release

**Comment by wolfv** [comment]
Hmm, it looks like there is another (packaging) issue with OS X. Is it possible that the RPath is not correctly set on installation (or something like this?)

**Comment by jwillemsen** [comment]
I don't have a MacOSX box to test, but I will open a new PR with an idea, not sure whether that works

**Comment by jwillemsen** [comment]
See https://github.com/traversaro/staged-recipes/pull/2

**Comment by jwillemsen** [comment]
Maybe ignore macosx for the moment

**Review by jwillemsen — APPROVED** [review]

**Comment by traversaro** [comment]
Thanks a lot @wolfv and @jwillemsen, I was away from keyboard in the last two days and it was really nice to see that the problem on these PR were fixed!

**Comment by jwillemsen** [comment]
It took some experiments but linux is now working stable. Got far with macosx but not 100% so disabled that for the moment.

**Pull request merged**